### PR TITLE
Expose the existing kinds matching args on Transactions query

### PR DESCRIPTION
**Pull request by znarf**
Part of https://github.com/opencollective/opencollective/issues/4590

### CODE REVIEW

File: server/graphql/v2/interface/IsMemberOf.js [hunk 1]
```diff
@@ -68,26 +68,28 @@ export const IsMemberOfFields = {
...
-          ],
-          group: ['role', 'collective.type'],
-          raw: true,
-        })
-      ).map(m => ({
-        role: m.role,
-        type: invert(AccountTypeToModelMapping)[m.type],
-      }));
+      // No await needed, GraphQL will take care of it
+      // TODO: try to skip if it's not a requested field
```
Comment (kewitz) [inline server/graphql/v2/interface/IsMemberOf.js:72]: 👍🏼

File: server/graphql/v2/interface/Collection.ts [hunk 2]
```diff
@@ -54,4 +54,12 @@ export interface CollectionReturnType {
   offset: number;
 }
 
+export interface TransactionsCollectionReturnType {
```
Comment (kewitz) [inline server/graphql/v2/interface/Collection.ts:57]: Extend `CollectionReturnType` with `kinds`.

**Review by kewitz — COMMENTED** [review]
💯

File: server/graphql/v2/interface/Collection.ts (hunk 2 above)
Comment (znarf) [reply server/graphql/v2/interface/Collection.ts:57]: @kewitz What would be the syntax?

File: server/graphql/v2/query/TransactionsQuery.ts [hunk 3]
```diff
@@ -152,6 +152,16 @@ const TransactionsQuery = {
     if (host) {
       where.push({ HostCollectiveId: host.id });
     }
+
+    // No await needed, GraphQL will take care of it
+    // TODO: try to skip if it's not a requested field
+    const existingKinds = models.Transaction.findAll({
+      attributes: ['kind'],
+      where,
```
Comment (kewitz) [inline server/graphql/v2/query/TransactionsQuery.ts:160]: Add where `kind IS NOT NULL`.
Screenshot from 2021-09-06 11-11-04

Comment (Betree) [reply server/graphql/v2/query/TransactionsQuery.ts:160]: Or a proper label in the frontend (eg. `Unknown`) if we still want the ability to filter them

**Pull request merged**